### PR TITLE
Remove all things from /srv/

### PR DESCRIPTION
**Pull request by ghost**
Follow up to #249. Addresses an error that occurred on subsequent deploys: `/srv/keys/ is not empty`

### CODE REVIEW

**Comment by dpoirier** [comment]
I'm slightly worried by this change. We don't create `/srv`, it's an existing directory, so it's possible some other software will put files there some day.

I wonder how hard it would be to move all of our data to a new subdirectory of `/srv` (e.g. `/srv/conf` or something) and then we could safely `rm -rf` it anytime we needed to.

Of course we'd need to make the change in a way that wouldn't break existing projects...

Another solution would be to just change it to `rm -rf /srv/salt /srv/pillar /srv/keys`.

**Comment by rebecca-caktus** [comment]
@poirier Good points. I liked the idea of moving to a subdirectory. Let me know what you think.

**Comment by rebecca-caktus** [comment]
Maybe `/srv/margarita` doesn't need to be moved to the subdirectory?

File: conf/salt/margarita.sls [hunk 1]
```diff
@@ -7,7 +7,7 @@ clone_repo:
      - name: git clone https://github.com/caktus/margarita.git margarita
      - user: root
      - unless: test -e /srv/margarita/.git
-     - cwd: /srv
+     - cwd: /srv/project
```
Comment (dpoirier) [inline conf/salt/margarita.sls]: We probably don't need this change.

File: fabfile.py [hunk 2]
```diff
@@ -157,8 +157,8 @@ def sync():
         salt_root = CONF_ROOT if CONF_ROOT.endswith('/') else CONF_ROOT + '/'
         project.rsync_project(
             local_dir=salt_root, remote_dir='/tmp/salt', delete=True)
-        sudo('rm -rf /srv/salt /srv/pillar')
-        sudo('mv /tmp/salt/* /srv/')
+        sudo('rm -rf /srv/project')
+        sudo('mv /tmp/salt/* /srv/project/')
```
Comment (dpoirier) [inline fabfile.py]: Don't we need to (re)create `/srv/project` before we move things into it?

**Comment by rebecca-caktus** [comment]
@dpoirier Updated.

File: conf/salt/margarita.sls [hunk 3]
```diff
@@ -7,7 +7,7 @@ clone_repo:
      - name: git clone https://github.com/caktus/margarita.git margarita
      - user: root
      - unless: test -e /srv/margarita/.git
-     - cwd: /srv
+     - cwd: /srv/
```
Comment (dpoirier) [inline conf/salt/margarita.sls]: Wow, does this really make a difference?

**Comment by dpoirier** [comment]
I just noticed that we're patching install_salt.sh. That's just a copy of the salt bootstrap script that we periodically download and save here for convenience, so any changes we make to our local copy won't persist.

We might not need to patch install_salt.sh.  It looks like our changes are in the FreeBSD install function.

If we do need to change its behavior, we will need to either pass arguments (https://docs.saltstack.com/en/latest/topics/tutorials/salt_bootstrap.html#command-line-options) or set environment variables (see the top of the script).  But I hope we don't need to do either.

File: fabfile.py [hunk 4]
```diff
@@ -157,8 +157,8 @@ def sync():
         salt_root = CONF_ROOT if CONF_ROOT.endswith('/') else CONF_ROOT + '/'
         project.rsync_project(
             local_dir=salt_root, remote_dir='/tmp/salt', delete=True)
-        sudo('rm -rf /srv/salt /srv/pillar')
-        sudo('mv /tmp/salt/* /srv/')
+        sudo('rm -rf /srv/project/*')
+        sudo('mv /tmp/salt/* /srv/project/')
```
Comment (dpoirier) [inline fabfile.py]: I'm not seeing where we create `/srv/project/` anywhere.  Is it getting created as a side effect of something else?

**Comment by dpoirier** [comment]
On reflection, clearly we'll need to make a change *somewhere* to tell Salt to look in /srv/projects. :-)

**Comment by rebecca-caktus** [comment]
I used this to set up a recent client project, and have submitted a couple more commits based on that experience.

**Comment by dpoirier** [comment]
I'm still concerned about patching our local copy of install_salt.sh.

**Comment by vkurup** [comment]
Should we go ahead with this change? I see 3 options:

1. Just remove the install_salt.sh changes from this PR (As Dan mentioned, that code path is only used when installing Salt on FreeBSD)
2. Decide not to add a `/srv/project` directory and instead add `rm -r /srv/keys` during each sync.
3. Something else...

(1) is somewhat cleaner, but (2) is easier to add to existing projects ("just add this 1 command to line NN in your fabfile")

Any preferences? I can update this PR with the desired change.

**Comment by dpoirier** [comment]
@vkurup at this point, (2) seems easiest to do and feel confident we're not breaking anything else.

**Comment by vkurup** [comment]
After merging in master, I realized that we've already made that change. The remaining change in this PR is to make sure that any deploy keys that we generate get placed in the 'keys' subdirectory alongside the GPG keys.

**Comment by dpoirier** [comment]
The remaining change doesn't seem to have anything to do with the original purpose of this pull request, does it?

**Comment by vkurup** [comment]
Only very tangentially... I think it belonged in #249. I've personally never used the `make staging-deploy-key` command, but if you did, then it would be placed in `conf/` and then when you sync, the command to move the files from `/tmp` to `/srv` would fail because of the presence of that file. (I think?)

**Comment by dpoirier** [comment]
Yeah, it looks like those changes should have been made to the Makefile along with #249.

:shark:

**Pull request merged**